File: invasion.py
class Invasion:
    """This class represents an invasion, and is initialized with
    data in JSON format

    """

    def __init__(self, data):

        self.id         = data['Id']
        self.node       = data['Node']
        self.planet     = data['Region']
        self.faction1   = data['InvaderInfo']['Faction']
        self.type1      = data['InvaderInfo']['MissionType']
        self.reward1    = data['InvaderInfo']['Reward']
        self.level1_min = data['InvaderInfo']['MinLevel']
        self.level1_max = data['InvaderInfo']['MaxLevel']
        self.faction2   = data['DefenderInfo']['Faction']
        self.type2      = data['DefenderInfo']['MissionType']
        self.reward2    = data['DefenderInfo']['Reward']
        self.level2_min = data['DefenderInfo']['MinLevel']
        self.level2_max = data['DefenderInfo']['MaxLevel']
        self.completion = data['Percentage']
        self.ETA        = data['Eta']
        self.desc       = data['Description']

    def __str__(self):
        """Returns a string with all the information about
        this invasion

        """
        if self.faction1 == 'Infestation':
            invasionString = ('{0} ({1})\n'
                              '{2} ({7})\n'
                              '{8}\n'
                              '{9:.2f}% - {10}')

        else:
            invasionString = ('{0} ({1}) - {2}\n'
                              '{3} ({4}, {5}) vs.\n'
                              '{6} ({7}, {8})\n'
                              '{9:.2f}% - {10}')

        return invasionString.format(self.node, self.planet, self.desc,
                                     self.faction1, self.type1,
                                     self.reward1, self.faction2,
                                     self.type2, self.reward2,
                                     self.completion, self.ETA)

    def get_rewards(self):
        """Returns a list containing the invasion's rewards excluding credits

        """
        return [i for i in [self.reward1, self.reward2] if 'cr' not in i]

File: scripts/lib/invasion.js
var util = require('util');
var dsUtil = require('./_utils.js');

var Invasion = function(data) {
  this.id = data.Id;
  this.node = data.Node;

  this.planet = data.Region; 
  this.faction1 = data.InvaderInfo.Faction;
  this.type1 = data.InvaderInfo.MissionType;
  this.reward1 = data.InvaderInfo.Reward;
  this.minLevel1 = data.InvaderInfo.MinLevel;
  this.maxLevel1 = data.InvaderInfo.MaxLevel;

  this.faction2 = data.DefenderInfo.Faction;
  this.type2 = data.DefenderInfo.MissionType;
  this.reward2 = data.DefenderInfo.Reward;
  this.minLevel2 = data.DefenderInfo.MinLevel;
  this.maxLevel2 = data.DefenderInfo.MaxLevel;

  this.completion = data.Percentage;
  this.ETA = data.Eta;
  this.desc = data.Description;
}

Invasion.prototype.toString = function() {
  if(this.faction1 === 'Infestation') {
    return util.format('%s (%s)\n' +
                       '%s (%s)\n' +
                       '%s\n' +
                       '%d% - %s',
                       this.node, this.planet, this.desc, this.type2,
                       this.reward2, Math.round(this.completion * 100) / 100,
                       this.ETA);
  }

  return util.format('%s (%s) - %s\n' +
                     '%s (%s, %s) vs.\n' +
                     '%s (%s, %s)\n' +
                     '%d% - %s',
                     this.node, this.planet, this.desc, this.faction1,
                     this.type1, this.reward1, this.faction2, this.type2,
                     this.reward2, Math.round(this.completion * 100) / 100, 
                     this.ETA);
}

// Only returns rewards that are not credits
Invasion.prototype.getRewards = function() {
  return [this.reward1, this.reward2].filter(function noCredits(reward) {
    return reward.indexOf('cr') !== -1
      ? true
      : false;
  });
}

module.exports = Invasion;
